packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -455,22 +455,23 @@ function measureBoxModelsMiddleware(): Middleware<Dispatch, State, Dispatch> {
     }
 }
 
+export interface IMessageChannel {
+  postMessage(message: any, transferables?: Transferable[]): void
+  set onmessage(handler: ((message: MessageEvent<any>) => any) | null)
+}
+
 export function messageChannelMiddleware(
   client: MakeswiftHostApiClient,
+  messageChannel?: IMessageChannel,
 ): Middleware<Dispatch, State, Dispatch> {
   return ({ dispatch, getState }: MiddlewareAPI<Dispatch, State>) =>
     (next: ReduxDispatch<Action>) => {
       let cleanUp = () => {}
-
-      if (typeof window === 'undefined') return cleanUp
-
-      const messageChannel = new window.MessageChannel()
-
-      window.parent.postMessage(messageChannel.port2, '*', [messageChannel.port2])
-
-      messageChannel.port1.onmessage = (event: MessageEvent<Action>) => dispatch(event.data)
+      if (typeof window === 'undefined' || !messageChannel) return cleanUp
```

Comment:
It's interesting we've got a `cleanUp` function we're returning but we're really not returning any clean up further down this function. It just looks like bad naming since a Redux middleware should be returning a dispatch function here, not a clean up function.

Reply:
I noticed that, too, will rename.